staNMF/staNMF.py

```diff
@@ -371,10 +392,16 @@ def instability(self, k1=0, k2=0):
                 self.instabilitydict[k] = (np.sum(distMat) / (numReplicates *
                                            (numReplicates-1)))
 
-                outputfile = open("instability.csv", "w")
-                for i in sorted(self.instabilitydict):
-                    outputfile.write(str(i) + "," +
-                                     str(self.instabilitydict[i]) + "\n")
+                if self.parallel:
```

Comment:
its not clear to me what this block is doing. If you set parallel, it saves each `k` iteratively? Is it writing the current `k` with the corresponding instability calculation? I think you can use `csv.writer` here (like you have in 402) instead of hardcoding the newline character